### ISSUE TITLE
Unable to remove shared file/folder

### ISSUE THREAD

**Issue by sjash** [nextcloud/server#14019]
### Steps to reproduce
1. Create a file/folder from a user.Go to sharing options and type the user/group you want to share with.
2. Un-share option doesn't appear for user who is sharing.
3. It appears for the users/group members to whom it is being shared

### Expected behaviour
If an user has shared an file/folder to any specific user/group/by email, then it should be able to revert in case of any mistakes.

### Actual behaviour
If an user has shared an file/folder to any specific user/group/by email un-share option doesn't appear for user who is sharing but appears for the user whom it is being shared.

### Server configuration

**Operating system**: Centos 7.2

**Web server:** Apache 2.4

**Database:** Mysql 5.6

**PHP version:** 7.0.6

**Nextcloud version:** 15.0.2

**Updated from an older Nextcloud/ownCloud or fresh install: Update from ver 14

**Where did you install Nextcloud from:**

**Signing status:**
<details>
<summary>Signing status</summary>

```
Login as admin user into your Nextcloud and access 
http://example.com/index.php/settings/integrity/failed 

No errors have been found.

```
</details>

**List of activated apps:**
<details>
<summary></summary>

```
If you have access to your command line run e.g.:
sudo -u www-data php occ app:list
from within your Nextcloud installation folder
 accessibility: 1.1.0
  - activity: 2.8.2
  - admin_audit: 1.5.0
  - bruteforcesettings: 1.3.0
  - checksum: 0.4.2
  - cloud_federation_api: 0.1.0
  - comments: 1.5.0
  - dav: 1.8.1
  - encryption: 2.3.0
  - federatedfilesharing: 1.5.0
  - federation: 1.5.0
  - files: 1.10.0
  - files_antivirus: 2.0.0
  - files_downloadactivity: 1.4.0
  - files_external: 1.6.0
  - files_pdfviewer: 1.4.0
  - files_rightclick: 0.11.0
  - files_sharing: 1.7.0
  - files_texteditor: 2.7.0
  - files_trashbin: 1.5.0
  - files_versions: 1.8.0
  - files_videoplayer: 1.4.0
  - firstrunwizard: 2.4.0
  - gallery: 18.2.0
  - groupfolders: 2.0.2
  - logreader: 2.0.0
  - lookup_server_connector: 1.3.0
  - metadata: 0.9.0
  - nextcloud_announcements: 1.4.0
  - notifications: 2.3.0
  - oauth2: 1.3.0
  - password_policy: 1.5.0
  - provisioning_api: 1.5.0
  - richdocuments: 3.1.1
  - serverinfo: 1.5.0
  - sharebymail: 1.5.0
  - socialsharing_email: 1.0.5
  - support: 1.0.0
  - survey_client: 1.3.0
  - systemtags: 1.5.0
  - tasks: 0.9.8
  - terms_of_service: 1.1.1
  - theming: 1.6.0
  - theming_customcss: 1.2.0
  - twofactor_backupcodes: 1.4.1
  - updatenotification: 1.5.0
  - user_ldap: 1.5.0
  - workflowengine: 1.5.0


</details>

**Nextcloud configuration:**
<details>
<summary>Config report</summary>

```
If you have access to your command line run e.g.:
sudo -u www-data php occ config:list system
from within your Nextcloud installation folder
"system": {
        "instanceid": "***REMOVED SENSITIVE VALUE***",
        "passwordsalt": "***REMOVED SENSITIVE VALUE***",
        "secret": "***REMOVED SENSITIVE VALUE***",
        "trusted_domains": [
            "cloud.example.in",
            "10.10.1.222",
            "10.10.1.111"
        ],
        "datadirectory": "***REMOVED SENSITIVE VALUE***",
        "overwrite.cli.url": "https:\/\/cloud.example.in",
        "dbtype": "mysql",
        "version": "15.0.2.0",
        "dbname": "***REMOVED SENSITIVE VALUE***",
        "dbhost": "***REMOVED SENSITIVE VALUE***",
        "dbport": "3306",
        "dbtableprefix": "oc_",
        "dbuser": "***REMOVED SENSITIVE VALUE***",
        "dbpassword": "***REMOVED SENSITIVE VALUE***",
        "logtimezone": "Asia\/Kolkata",
        "installed": true,
        "ldapIgnoreNamingRules": false,
        "ldapProviderFactory": "\\OCA\\User_LDAP\\LDAPProviderFactory",
        "auth.bruteforce.protection.enabled": false,
        "trusted_proxies": "***REMOVED SENSITIVE VALUE***",
        "mail_smtpmode": "smtp",
        "mail_from_address": "***REMOVED SENSITIVE VALUE***",
        "mail_domain": "***REMOVED SENSITIVE VALUE***",
        "enable_previews": true,
        "preview_max_scale_factor": 10,
        "ppstoreurlpreview_libreoffice_path": "\/usr\/bin\/libreoffice",
        "preview_office_cl_parameters": " --headless --nologo --nofirststartwizard --invisible --norestore -convert-to pdf -outdir ",
        "loglevel": 2,
        "logfile": "\/mnt\/cloud-data\/nextcloud.log",
        "enable_avatars": true,
        "knowledgebaseenabled": false,
        "appcodechecker": true,
        "lost_password_link": "https:\/\/auth.example.in\/service\/?action=sendtoken",
        "enable_certificate_management": false,
        "memcache.local": "\\OC\\Memcache\\APCu",
        "maintenance": false,
        "mail_smtpauthtype": "PLAIN",
        "mail_smtphost": "***REMOVED SENSITIVE VALUE***",
        "mail_smtpport": "465",
        "mail_smtpdebug": true,
        "mail_smtpauth": 1,
        "mail_smtpsecure": "ssl",
        "mail_smtpname": "***REMOVED SENSITIVE VALUE***",
        "mail_smtppassword": "***REMOVED SENSITIVE VALUE***",
        "updater.release.channel": "stable",
        "log_rotate_size": 52428800,
        "session_lifetime": 1800
    }
</details>

**Are you using external storage, if yes which one:** local/smb/sftp/ -- YES

**Are you using encryption:** NO

**Are you using an external user-backend, if yes which one:** LDAP

### Client configuration
**Browser:** Firefox  v64/Chrome 71

**Operating system:** Windows/ Ubuntu/Centos

**Comment by mickymadsystems** [comment]
I had the same problems and I solved it by deactivating the application Social sharing via email.

**Comment by violoncelloCH** [comment]
that's a bug which should be fixed by #13738 in NC15.0.3 see [nextcloud/socialsharing#29 (comment)](https://github.com/nextcloud/socialsharing/issues/29#issuecomment-456421423)
